feat: middleware.ts created for protect the routes

black-arm opened the pull request.

Balastrong commented:
Thank you for the PR!

When I navigate to a protected page I now get this error, we should probably make sure it redirects to the homepage. Any idea?

![image](https://github.com/Balastrong/github-stats/assets/7253929/6a4b285a-26b5-4986-b2a8-f62f3fcb6356)

black-arm commented:
the redirect work if we deleted secret variable from [...nextauth].ts and in .env.local  we added the NEXTAUTH_SECRET

<img width="1306" alt="Screenshot 2023-08-25 alle 09 58 04" src="https://github.com/Balastrong/github-stats/assets/68558867/ac18af9e-6bb6-4362-b487-0dedb51c68ee">
<img width="552" alt="Screenshot 2023-08-25 alle 10 01 40" src="https://github.com/Balastrong/github-stats/assets/68558867/35837841-a7ed-46c1-be6e-8352e1002a5b">
I check 

how do we want to proceed? Do we update the .env and documentation or do we find to way to work with secret variable

Balastrong commented:
Finding a way that does not require an extra variable would be great :)

Balastrong commented:
Still thinking about it... we might entirely replace `SECRET` with `NEXTAUTH_SECRET` and leave a value there already in `.env.example` to keep onboarding easy

Balastrong commented:
Exactly, and you can also give a default value to `NEXTAUTH_SECRET` in `.env.example` so that it just works :)

Balastrong commented on .env.example:
```diff
@@ -6,4 +6,4 @@ DEV_GITHUB_TOKEN=
 #GITHUB_SECRET=
 
 # Next-Auth secret (not required for local development)
```
The description should be updated too as now it's actually required to make it work

Balastrong commented on .env.example:
```diff
@@ -6,4 +6,4 @@ DEV_GITHUB_TOKEN=
 # Next-Auth secret (not required for local development)
-#SECRET=
\ No newline at end of file
+#NEXTAUTH_SECRET=oYoOxbxGQmKOqbJmxf5h1RScYrC8DZ2BgL2OxT5w/C8=
```
Since it is now mandatory, I'd remove the comment `#` so it can already be used

Balastrong reviewed (changes requested):
Thank you!

I added a couple comments to make sure the onboarding remains correct and seamless for new contributors

Balastrong reviewed (approved):
LGTM! Thank you! 🚀
